File: src/main/kotlin/easy/TwoSum.kt
fun twoSum(
    nums: IntArray,
    target: Int,
): IntArray {
    val seen = mutableMapOf<Int, Int>()
    for ((index, num) in nums.withIndex()) {
        val complement = seen.get(target - num)
        if (complement != null) {
            return intArrayOf(complement, index)
        }
        seen[num] = index
    }
    return intArrayOf()
}

File: lib/easy/two_sum.dart
List<int> twoSum(List<int> nums, int target) {
  var seen = <int, int>{};
  int index = 0;
  for (int num in nums) {
    int complement = target - num;
    if (seen.containsKey(complement)) {
      return [seen[complement] ?? 0, index];
    }
    seen[num] = index;
    index += 1;
  }
  return [];
}
